compiler/src/org.graalvm.compiler.truffle.compiler/src/org/graalvm/compiler/truffle/compiler/phases/inlining/DefaultInliningPolicy.java

```diff
@@ -128,6 +132,18 @@ private void inline(CallTree tree) {
         }
     }
 
+    private boolean isExcludedMethod(CallNode candidate, String[] excludedMethods) {
+        for (String method : excludedMethods) {
+            if (method.isEmpty()) {
+                continue;
+            }
+            if (candidate.getTruffleAST().getName().contains(method)) {
```

Comment:
Hm... the `contains` here might be an issue as it's a bit too aggressive (e.g. `ExcludeInlining=init` would exclude `init` `initialize` `__init__`). 
The annoying part is that as of late `getName` also includes splitting info (see org.graalvm.compiler.truffle.runtime.OptimizedCallTarget#getName e.g. `foo <split-12312>` )  so using `equals` will miss any splits :/ 
The correct thing to do would be to expose the raw name of the root node through the call target but that might be overkill for this PR so maybe just explicitly handle (i.e. remove) the `<split-xxxxx>` pattern if appears in the name than use `equals`

Reply:
Sounds good. I can remove the split pattern and then use `equals`, which then means having to pass in both the class and method, e.g. `Object#foo` instead of just the method `foo`.

If that's intended, should we update the description (or maybe give an example) or do we assume people have that knowledge already?

Reply:
In any case... [`92ae6d2` (#3524)](https://github.com/oracle/graal/pull/3524/commits/92ae6d24558b9976540f19681cf35bb1e40de5b3)

Reply:
I think it's good to provide the full name of the call target one want's to exclude. e.g. Consider the java `toString` method, running with `ExcludeInlining=toString` will very likely match more than one wants for the experiment while going `ExcludeInlining=MyExperimentClass#toString` (or whatever the syntax) would be correct.